app/controllers/navigation/Navigator.scala

```diff
@@ -329,10 +327,10 @@ object Navigator {
 
   private def representativeAgentClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
     if (cacheModel.isExs) {
-      if (cacheModel.parties.consignorDetails.flatMap(_.details.eori.map(_.value)).getOrElse("").nonEmpty)
-        controllers.declaration.routes.ConsignorEoriNumberController.displayPage(mode)
-      else
+      if (cacheModel.isExs && cacheModel.parties.consignorDetails.flatMap(_.details.address).isDefined)
```

Comment:
Isn't this check redundant?
Why could you just have 
```
if (cacheModel.parties.consignorDetails.flatMap(_.details.address).isDefined)
```

You did a similar thing on line 312

Reply:
You check `cacheModel.isExs` in line 329 so in line 330 the first part of the check is unnecessary

Reply:
👍